prac_09/sort_files_1.py

```diff
@@ -0,0 +1,46 @@
+import os
+import shutil
+
+
+TARGET_FILE = 'FilesToSort'
+
+
+def main():
+
+    os.chdir(TARGET_FILE)
+
+    print('Sorting files in: {}'.format(TARGET_FILE))
+    for directory_name, subdirectories, filenames in os.walk('.'):
+        type_list = create_type_list(filenames)
+        for ext_type in type_list:
+            try:
+                os.mkdir(ext_type)
+            except FileExistsError:
+                pass
+        try:
+            for file in filenames:
+                file_ext = get_extension(file)
+                shutil.move(file, file_ext)
+        except shutil.Error:
+            pass
+    print('Sort Complete')
+
+
+def create_type_list(filenames):
+    """Create a list of the different types of extensions in the target file."""
+    type_list = []
+    for file in filenames:
+        file_ext = get_extension(file).replace('.', '')
+        if file_ext not in type_list:
+            type_list.append(file_ext)
+    return type_list
+
+
+def get_extension(file):
+    """Get the extension of target file."""
+    file_type = file.split('.')[-1]
```

Comment:
You can actually use this to get extensions:
for filename in os.listdir('.'):
    extension = os.path.splitext(filename)[1][1:]

It will loop through each item and get the extension. But your method also works well :)